### PR TITLE
fix(question): surface pending prompts hidden by the TUI

### LINKED ISSUE #7338: [Bug]: Interactive `question` tool prompt never renders in TUI when parent turn continues after a sync subagent task completes
### Prerequisites

- [x] I will write this issue in English (see our [Language Policy](https://github.com/code-yeongyu/oh-my-openagent/blob/dev/CONTRIBUTING.md#language-policy))
- [x] I have searched existing issues to avoid duplicates
- [x] I am using the latest version of oh-my-openagent
- [x] I have read the [documentation](https://github.com/code-yeongyu/oh-my-openagent#readme) or asked an AI coding agent with this project's GitHub URL loaded and couldn't find the answer

## Bug Description

When a Sisyphus parent turn continues **after a synchronous `task()` delegation completes**, any subsequent `question` tool call executes and blocks waiting for user input, but the interactive prompt is **never rendered in the TUI**. The session then appears completely stuck: no visible progress, no error, and no way to know an answer is expected.

Answering as a plain text chat message unblocks the pending ask state immediately, so the tool itself works — only the UI rendering is missing.

Possibly related to #5435 (internal-dispatch vs UI-rendering family of issues).

## Environment

-

### Steps to Reproduce

1. Configure OMO on OpenCode (Windows, PowerShell shell) with `runtime_fallback` enabled.
2. In a Sisyphus primary session, trigger a sync delegation: `task(category=..., ...)` spawning a Sisyphus-Junior.
3. Wait for the child to finish its loop ("exiting loop" → session cancelled).
4. Let the parent continue its turn autonomously (ingesting the child result) until it decides to call the `question` tool.
5. Observe the TUI.

### Expected Behavior

The interactive question (options + custom input) renders in the TUI and waits for user selection.

### Actual Behavior

- The assistant message ends with a `[tool: question]` part (visible in storage/transcript).
- The provider stream for that message starts but never completes; **no `session.idle` event is emitted** afterwards.
- Zero errors logged. The session looks frozen from the user's perspective.
- Typing a plain text reply in the same session resolves the pending ask and work resumes normally.

### Doctor Output

```shell
PS > bunx oh-my-openagent version
oh-my-openagent v4.19.4


PS > opencode --version
1.18.23

PS > bunx oh-my-openagent doctor

 oMoMoMoMo Doctor

 ⚠ 34 issues found:

1. Deprecated reasoning config key
   C:\Users\jc_lo\.omo\omo.jsonc: [opencode].agents.sisyphus.fallback_models
   Fix: Replace fallback_models with models, or run: oh-my-openagent config migrate
   Affects: [opencode].agents.sisyphus.fallback_models

2. Deprecated reasoning config key
   C:\Users\jc_lo\.omo\omo.jsonc: [opencode].agents.hephaestus.variant
   Fix: Replace variant with reasoning, or run: oh-my-openagent config migrate
   Affects: [opencode].agents.hephaestus.variant

3. Deprecated reasoning config key
   C:\Users\jc_lo\.omo\omo.jsonc: [opencode].agents.hephaestus.fallback_models
   Fix: Replace fallback_models with models, or run: oh-my-openagent config migrate
   Affects: [opencode].agents.hephaestus.fallback_models

4. Deprecated reasoning config key
   C:\Users\jc_lo\.omo\omo.jsonc: [opencode].agents.oracle.variant
   Fix: Replace variant with reasoning, or run: oh-my-openagent config migrate
   Affects: [opencode].agents.oracle.variant

5. Deprecated reasoning config key
   C:\Users\jc_lo\.omo\omo.jsonc: [opencode].agents.oracle.fallback_models
   Fix: Replace fallback_models with models, or run: oh-my-openagent config migrate
   Affects: [opencode].agents.oracle.fallback_models

6. Deprecated reasoning config key
   C:\Users\jc_lo\.omo\omo.jsonc: [opencode].agents.librarian.fallback_models
   Fix: Replace fallback_models with models, or run: oh-my-openagent config migrate
   Affects: [opencode].agents.librarian.fallback_models

7. Deprecated reasoning config key
   C:\Users\jc_lo\.omo\omo.jsonc: [opencode].agents.explore.fallback_models
   Fix: Replace fallback_models with models, or run: oh-my-openagent config migrate
   Affects: [opencode].agents.explore.fallback_models

8. Deprecated reasoning config key
   C:\Users\jc_lo\.omo\omo.jsonc: [opencode].agents.multimodal-looker.variant
   Fix: Replace variant with reasoning, or run: oh-my-openagent config migrate
   Affects: [opencode].agents.multimodal-looker.variant

9. Deprecated reasoning config key
   C:\Users\jc_lo\.omo\omo.jsonc: [opencode].agents.multimodal-looker.fallback_models
   Fix: Replace fallback_models with models, or run: oh-my-openagent config migrate
   Affects: [opencode].agents.multimodal-looker.fallback_models

10. Deprecated reasoning config key
   C:\Users\jc_lo\.omo\omo.jsonc: [opencode].agents.prometheus.variant
   Fix: Replace variant with reasoning, or run: oh-my-openagent config migrate
   Affects: [opencode].agents.prometheus.variant

11. Deprecated reasoning config key
   C:\Users\jc_lo\.omo\omo.jsonc: [opencode].agents.prometheus.fallback_models
   Fix: Replace fallback_models with models, or run: oh-my-openagent config migrate
   Affects: [opencode].agents.prometheus.fallback_models

12. Deprecated reasoning config key
   C:\Users\jc_lo\.omo\omo.jsonc: [opencode].agents.metis.variant
   Fix: Replace variant with reasoning, or run: oh-my-openagent config migrate
   Affects: [opencode].agents.metis.variant

13. Deprecated reasoning config key
   C:\Users\jc_lo\.omo\omo.jsonc: [opencode].agents.metis.fallback_models
   Fix: Replace fallback_models with models, or run: oh-my-openagent config migrate
   Affects: [opencode].agents.metis.fallback_models

14. Deprecated reasoning config key
   C:\Users\jc_lo\.omo\omo.jsonc: [opencode].agents.momus.variant
   Fix: Replace variant with reasoning, or run: oh-my-openagent config migrate
   Affects: [opencode].agents.momus.variant

15. Deprecated reasoning config key
   C:\Users\jc_lo\.omo\omo.jsonc: [opencode].agents.momus.fallback_models
   Fix: Replace fallback_models with models, or run: oh-my-openagent config migrate
   Affects: [opencode].agents.momus.fallback_models

16. Deprecated reasoning config key
   C:\Users\jc_lo\.omo\omo.jsonc: [opencode].agents.atlas.fallback_models
   Fix: Replace fallback_models with models, or run: oh-my-openagent config migrate
   Affects: [opencode].agents.atlas.fallback_models

17. Deprecated reasoning config key
   C:\Users\jc_lo\.omo\omo.jsonc: [opencode].agents.sisyphus-junior.fallback_models
   Fix: Replace fallback_models with models, or run: oh-my-openagent config migrate
   Affects: [opencode].agents.sisyphus-junior.fallback_models

18. Deprecated reasoning config key
   C:\Users\jc_lo\.omo\omo.jsonc: [opencode].categories.visual-engineering.variant
   Fix: Replace variant with reasoning, or run: oh-my-openagent config migrate
   Affects: [opencode].categories.visual-engineering.variant

19. Deprecated reasoning config key
   C:\Users\jc_lo\.omo\omo.jsonc: [opencode].categories.visual-engineering.fallback_models
   Fix: Replace fallback_models with models, or run: oh-my-openagent config migrate
   Affects: [opencode].categories.visual-engineering.fallback_models

20. Deprecated reasoning config key
   C:\Users\jc_lo\.omo\omo.jsonc: [opencode].categories.ultrabrain.variant
   Fix: Replace variant with reasoning, or run: oh-my-openagent config migrate
   Affects: [opencode].categories.ultrabrain.variant

21. Deprecated reasoning config key
   C:\Users\jc_lo\.omo\omo.jsonc: [opencode].categories.ultrabrain.fallback_models
   Fix: Replace fallback_models with models, or run: oh-my-openagent config migrate
   Affects: [opencode].categories.ultrabrain.fallback_models

22. Deprecated reasoning config key
   C:\Users\jc_lo\.omo\omo.jsonc: [opencode].categories.deep.variant
   Fix: Replace variant with reasoning, or run: oh-my-openagent config migrate
   Affects: [opencode].categories.deep.variant

23. Deprecated reasoning config key
   C:\Users\jc_lo\.omo\omo.jsonc: [opencode].categories.deep.fallback_models
   Fix: Replace fallback_models with models, or run: oh-my-openagent config migrate
   Affects: [opencode].categories.deep.fallback_models

24. Deprecated reasoning config key
   C:\Users\jc_lo\.omo\omo.jsonc: [opencode].categories.artistry.variant
   Fix: Replace variant with reasoning, or run: oh-my-openagent config migrate
   Affects: [opencode].categories.artistry.variant

25. Deprecated reasoning config key
   C:\Users\jc_lo\.omo\omo.jsonc: [opencode].categories.artistry.fallback_models
   Fix: Replace fallback_models with models, or run: oh-my-openagent config migrate
   Affects: [opencode].categories.artistry.fallback_models

26. Deprecated reasoning config key
   C:\Users\jc_lo\.omo\omo.jsonc: [opencode].categories.quick.fallback_models
   Fix: Replace fallback_models with models, or run: oh-my-openagent config migrate
   Affects: [opencode].categories.quick.fallback_models

27. Deprecated reasoning config key
   C:\Users\jc_lo\.omo\omo.jsonc: [opencode].categories.unspecified-low.variant
   Fix: Replace variant with reasoning, or run: oh-my-openagent config migrate
   Affects: [opencode].categories.unspecified-low.variant

28. Deprecated reasoning config key
   C:\Users\jc_lo\.omo\omo.jsonc: [opencode].categories.unspecified-low.fallback_models
   Fix: Replace fallback_models with models, or run: oh-my-openagent config migrate
   Affects: [opencode].categories.unspecified-low.fallback_models

29. Deprecated reasoning config key
   C:\Users\jc_lo\.omo\omo.jsonc: [opencode].categories.unspecified-high.variant
   Fix: Replace variant with reasoning, or run: oh-my-openagent config migrate
   Affects: [opencode].categories.unspecified-high.variant

30. Deprecated reasoning config key
   C:\Users\jc_lo\.omo\omo.jsonc: [opencode].categories.unspecified-high.fallback_models
   Fix: Replace fallback_models with models, or run: oh-my-openagent config migrate
   Affects: [opencode].categories.unspecified-high.fallback_models

31. Deprecated reasoning config key
   C:\Users\jc_lo\.omo\omo.jsonc: [opencode].categories.writing.variant
   Fix: Replace variant with reasoning, or run: oh-my-openagent config migrate
   Affects: [opencode].categories.writing.variant

32. Deprecated reasoning config key
   C:\Users\jc_lo\.omo\omo.jsonc: [opencode].categories.writing.fallback_models
   Fix: Replace fallback_models with models, or run: oh-my-openagent config migrate
   Affects: [opencode].categories.writing.fallback_models

33. GitHub CLI missing
   gh CLI is not installed.
   Fix: Install from https://cli.github.com/
   Affects: GitHub automation

34. Configured models rely on compatibility fallback
   librarian=opencode-go/ox-alpha-free (unknown), explore=opencode-go/ox-alpha-free (unknown), quick=opencode-go/ox-alpha-free (unknown)
   Affects: librarian, explore, quick
```

### Error Logs

```shell

```

### Configuration

```json
// OMO configuration
{
  "[opencode]": {
    "$schema": "https://raw.githubusercontent.com/code-yeongyu/oh-my-openagent/dev/assets/omo.schema.json",
    "agents": {
      "sisyphus": {
        "model": "opencode-go/glm-5.3",
        "fallback_models": [
          {
            "model": "openai/gpt-5.6-sol",
            "variant": "medium"
          },
          {
            "model": "openrouter/z-ai/glm-5.3"
          },
          {
            "model": "minimax-coding-plan/MiniMax-M3"
          }
        ]
      },
      "hephaestus": {
        "model": "openai/gpt-5.6-sol",
        "variant": "medium",
        "fallback_models": [
          {
            "model": "openrouter/openai/gpt-5.6-sol"
          },
          {
            "model": "minimax-coding-plan/MiniMax-M3"
          },
          {
            "model": "opencode-go/mimo-v2.5-free"
          }
        ]
      },
      "oracle": {
        "model": "openai/gpt-5.6-sol",
        "variant": "xhigh",
        "fallback_models": [
          {
            "model": "opencode-go/glm-5.3"
          },
          {
            "model": "openrouter/openai/gpt-5.6-sol"
          },
          {
            "model": "minimax-coding-plan/MiniMax-M3"
          },
          {
            "model": "opencode-go/mimo-v2.5-free"
          }
        ]
      },
      "librarian": {
        "model": "opencode-go/ox-alpha-free",
        "fallback_models": [
          {
            "model": "openai/gpt-5.6-luna-fast",
            "variant": "low"
          },
          {
            "model": "opencode-go/qwen3.7-plus"
          },
          {
            "model": "opencode-go/minimax-m3"
          },
          {
            "model": "minimax-coding-plan/MiniMax-M3"
          },
          {
            "model": "opencode-go/minimax-m2.7"
          },
          {
            "model": "openai/gpt-5.4-nano"
          },
          {
            "model": "openrouter/deepseek/deepseek-v4-flash"
          }
        ]
      },
      "explore": {
        "model": "opencode-go/ox-alpha-free",
        "fallback_models": [
          {
            "model": "openai/gpt-5.6-luna-fast",
            "variant": "low"
          },
          {
            "model": "opencode-go/qwen3.7-plus"
          },
          {
            "model": "opencode-go/minimax-m3"
          },
          {
            "model": "minimax-coding-plan/MiniMax-M3"
          },
          {
            "model": "opencode-go/minimax-m2.7"
          },
          {
            "model": "openai/gpt-5.4-nano"
          },
          {
            "model": "openrouter/deepseek/deepseek-v4-flash"
          }
        ]
      },
      "multimodal-looker": {
        "model": "openai/gpt-5.6-sol",
        "variant": "low",
        "fallback_models": [
          {
            "model": "opencode-go/glm-5.3"
          },
          {
            "model": "openai/gpt-5-nano"
          },
          {
            "model": "openrouter/deepseek/deepseek-v4-flash"
          }
        ]
      },
      "prometheus": {
        "model": "opencode-go/glm-5.3",
        "variant": "high",
        "fallback_models": [
          {
            "model": "openrouter/z-ai/glm-5.3"
          },
          {
            "model": "minimax-coding-plan/MiniMax-M3"
          },
          {
            "model": "opencode-go/mimo-v2.5-free"
          }
        ]
      },
      "metis": {
        "model": "opencode-go/glm-5.3",
        "variant": "low",
        "fallback_models": [
          {
            "model": "openrouter/z-ai/glm-5.3"
          },
          {
            "model": "minimax-coding-plan/MiniMax-M3"
          },
          {
            "model": "opencode-go/mimo-v2.5-free"
          }
        ]
      },
      "momus": {
        "model": "openai/gpt-5.6-terra",
        "variant": "high",
        "fallback_models": [
          {
            "model": "openai/gpt-5.6-sol",
            "variant": "xhigh"
          },
          {
            "model": "opencode-go/glm-5.3"
          },
          {
            "model": "openrouter/openai/gpt-5.6-sol"
          }
        ]
      },
      "atlas": {
        "model": "opencode-go/glm-5.3",
        "fallback_models": [
          {
            "model": "openai/gpt-5.6-sol",
            "variant": "medium"
          },
          {
            "model": "opencode-go/minimax-m3"
          },
          {
            "model": "minimax-coding-plan/MiniMax-M3"
          },
          {
            "model": "opencode-go/minimax-m2.7"
          },
          {
            "model": "openrouter/z-ai/glm-5.3"
          }
        ]
      },
      "sisyphus-junior": {
        "model": "opencode-go/glm-5.3",
        "prompt_append": "[Shell safety - Windows PowerShell] Every shell call needs an explicit timeout param: short reads 30s, focused build/test 120s, full suite 300s max. For dotnet test always add --blame-hang --blame-hang-timeout 3m. Never launch servers or dotnet run in foreground; if a command times out, diagnose before retrying the same command. Set MSBUILDDISABLENODEREUSE=1 before piped dotnet build/test commands (node-reuse workers inherit pipe handles and hang pipelines after tests finish). Before any long-running command, load the skill safe-powershell-execution.",
        "fallback_models": [
          {
            "model": "openai/gpt-5.6-sol",
            "variant": "medium"
          },
          {
            "model": "opencode-go/minimax-m3"
          },
          {
            "model": "minimax-coding-plan/MiniMax-M3"
          },
          {
            "model": "opencode-go/minimax-m2.7"
          },
          {
            "model": "openrouter/z-ai/glm-5.3"
          }
        ]
      }
    },
    "categories": {
      "visual-engineering": {
        "model": "opencode-go/glm-5.3",
        "variant": "high",
        "prompt_append": "[Shell safety - Windows PowerShell] Every shell call needs an explicit timeout param: short reads 30s, focused build/test 120s, full suite 300s max. Never launch dev servers or dotnet run in foreground; use one bounded invocation with Start-Process + redirects + readiness deadline + cleanup in finally. If a command times out, diagnose before retrying the same command. Set MSBUILDDISABLENODEREUSE=1 before piped dotnet build/test commands (node-reuse workers inherit pipe handles and hang pipelines after tests finish). Before any long-running command, load the skill safe-powershell-execution.",
        "fallback_models": [
          {
            "model": "opencode-go/glm-5.3",
            "variant": "high"
          },
          {
            "model": "openai/gpt-5.6-sol",
            "variant": "medium"
          },
          {
            "model": "openrouter/z-ai/glm-5.3"
          }
        ]
      },
      "ultrabrain": {
        "model": "openai/gpt-5.6-sol",
        "variant": "xhigh",
        "fallback_models": [
          {
            "model": "openrouter/openai/gpt-5.6-sol"
          },
          {
            "model": "minimax-coding-plan/MiniMax-M3"
          },
          {
            "model": "opencode-go/mimo-v2.5-free"
          }
        ]
      },
      "deep": {
        "model": "openai/gpt-5.6-sol",
        "variant": "medium",
        "prompt_append": "[Shell safety - Windows PowerShell] Every shell call needs an explicit timeout param: short reads 30s, focused build/test 120s, full suite 300s max. For dotnet test always add --blame-hang --blame-hang-timeout 3m. Never launch servers in foreground; if a command times out, diagnose before retrying the same command. Set MSBUILDDISABLENODEREUSE=1 before piped dotnet build/test commands (node-reuse workers inherit pipe handles and hang pipelines after tests finish). Before any long-running command, load the skill safe-powershell-execution.",
        "fallback_models": [
          {
            "model": "openrouter/openai/gpt-5.6-sol"
          },
          {
            "model": "minimax-coding-plan/MiniMax-M3"
          },
          {
            "model": "opencode-go/mimo-v2.5-free"
          }
        ]
      },
      "artistry": {
        "model": "opencode-go/glm-5.3",
        "variant": "high",
        "fallback_models": [
          {
            "model": "openrouter/z-ai/glm-5.3"
          },
          {
            "model": "minimax-coding-plan/MiniMax-M3"
          },
          {
            "model": "opencode-go/mimo-v2.5-free"
          }
        ]
      },
      "quick": {
        "model": "opencode-go/ox-alpha-free",
        "prompt_append": "[Shell safety - Windows PowerShell] Every shell call needs an explicit timeout param: short reads 30s, focused build/test 120s max. For dotnet test always add --blame-hang --blame-hang-timeout 3m. If a command times out, diagnose before retrying the same command. Set MSBUILDDISABLENODEREUSE=1 before piped dotnet build/test commands (node-reuse workers inherit pipe handles and hang pipelines after tests finish). Before any long-running command, load the skill safe-powershell-execution.",
        "fallback_models": [
          {
            "model": "opencode-go/qwen3.6-flash"
          },
          {
            "model": "opencode-go/minimax-m3",
            "variant": "high"
          },
          {
            "model": "opencode-go/minimax-m2.7",
            "variant": "high"
          },
          {
            "model": "openrouter/qwen/qwen3.6-flash"
          }
        ]
      },
      "unspecified-low": {
        "model": "openai/gpt-5.6-terra",
        "variant": "high",
        "prompt_append": "[Shell safety - Windows PowerShell] Every shell call needs an explicit timeout param: short reads 30s, focused build/test 120s, full suite 300s max. For dotnet test always add --blame-hang --blame-hang-timeout 3m. Never launch servers in foreground; if a command times out, diagnose before retrying the same command. Set MSBUILDDISABLENODEREUSE=1 before piped dotnet build/test commands (node-reuse workers inherit pipe handles and hang pipelines after tests finish). Before any long-running command, load the skill safe-powershell-execution.",
        "fallback_models": [
          {
            "model": "opencode-go/deepseek-v4-pro",
            "variant": "max"
          },
          {
            "model": "opencode-go/mimo-v2.5-pro"
          },
          {
            "model": "openrouter/openai/gpt-5.6-sol"
          }
        ]
      },
      "unspecified-high": {
        "model": "openai/gpt-5.6-terra",
        "variant": "high",
        "prompt_append": "[Shell safety - Windows PowerShell] Every shell call needs an explicit timeout param: short reads 30s, focused build/test 120s, full suite 300s max. For dotnet test always add --blame-hang --blame-hang-timeout 3m. Never launch servers in foreground; if a command times out, diagnose before retrying the same command. Set MSBUILDDISABLENODEREUSE=1 before piped dotnet build/test commands (node-reuse workers inherit pipe handles and hang pipelines after tests finish). Before any long-running command, load the skill safe-powershell-execution.",
        "fallback_models": [
          {
            "model": "opencode-go/deepseek-v4-pro",
            "variant": "max"
          },
          {
            "model": "opencode-go/mimo-v2.5-pro"
          },
          {
            "model": "openrouter/openai/gpt-5.6-sol"
          }
        ]
      },
      "writing": {
        "model": "opencode-go/glm-5.3",
        "variant": "low",
        "fallback_models": [
          {
            "model": "openrouter/z-ai/glm-5.3"
          },
          {
            "model": "minimax-coding-plan/MiniMax-M3"
          },
          {
            "model": "opencode-go/mimo-v2.5-free"
          }
        ]
      }
    },
    "runtime_fallback": {
      "enabled": true,
      "retry_on_errors": [400, 401, 403, 404, 429, 500, 502, 503, 504],
      "max_fallback_attempts": 3,
      "cooldown_seconds": 60,
      "timeout_seconds": 0,
      "notify_on_fallback": true
    }
  }
}
```

### Additional Context

 OS: Windows 11 (win32), PowerShell shell
- opencode: 1.18.22 → 1.18.23 (observed across restart)
- oh-my-openagent: 4.19.4
- Model serving the parent turn: minimax-coding-plan/MiniMax-M3 (after quota fallback) — likely model-independent

- Understood and **not** part of this report: subagent sessions are intentionally created with `QUESTION_DENIED_SESSION_PERMISSION` (`{permission:"question", action:"deny", pattern:"*"}`) — workers correctly escalate via final reports instead of asking.
- The failing case is the **parent** session whose turn originates from OMO's internal dispatch chain (post-child continuation via promptAsync machinery) rather than a direct user message. Hypothesis: interactive prompts attached to internally-dispatched turns are not surfaced by the TUI.
- Workaround for affected users: reply to the invisible question as normal chat text.

### Operating System

Windows

### OpenCode Version

1.18.22

### PR DESCRIPTION
**Pull request by AceRothstein71**
Fixes #7338

## Root cause
The OpenCode TUI renders questions from a one-shot SSE event, does not rehydrate pending questions after reconnect, and shadows the question prompt while permissions are pending. Dense sync-task completion traffic makes the loss window observable even though the server-side question remains valid.

## Changes
- Add a question visibility watchdog at the existing question execution seam
- Probe the real tool-part state after 30 seconds
- Emit a delivery-independent TUI toast when the question is still waiting for the user
- Preserve directory scoping and avoid duplicate notification after completion

## Validation
- Watchdog and handler tests: 30/30
- omo-opencode tsgo: exit 0, repeated three times
- Live isolated OpenCode probe observed the watchdog toast on the event wire, then reply and idle
- Creator full-vertical audit: two consecutive zero-finding waves
- Evidence: `.omo/evidence/20260826-issue7338-question-render/`

## Residual risk
Interactive TUI driving was unavailable because tmux is absent. The wire-level toast assertion covers the changed delivery path, but final terminal rendering remains an upstream TUI concern.

### CODE REVIEW

File: packages/omo-opencode/src/features/question-visibility-watchdog/watchdog.ts [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+      const result = await deps.probeQuestionToolState(ref)
+      if (result.state !== "waiting") {
+        log(`${WATCHDOG_LOG_PREFIX} Question resolved before grace elapsed`, {
+          sessionID: ref.sessionID,
+          callID: ref.callID,
+          detail: JSON.stringify(result),
+        })
+        return
+      }
+      await deps.showToast(buildToastBody(ref))
```
Comment (code-yeongyu) [inline packages/omo-opencode/src/features/question-visibility-watchdog/watchdog.ts:61]: [sisyphus-bot] `runCheck` has no disposed/generation check after the awaited probe. If plugin shutdown/reload happens while `session.messages` is in flight, this stale callback can still call `showToast` after disposal. Wire `dispose()` into the plugin lifecycle and reject callbacks started by an older generation before emitting UI.

File: packages/omo-opencode/src/features/question-visibility-watchdog/watchdog.ts [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    if (disposed) return
+    if (!ref.sessionID) return
+    const key = refKey(ref)
+    if (scheduled.has(key)) return
+    log(`${WATCHDOG_LOG_PREFIX} Watching question execution`, {
+      sessionID: ref.sessionID,
+      callID: ref.callID,
+      graceMs,
+    })
+    const cancel = schedule(() => {
```
Comment (code-yeongyu) [inline packages/omo-opencode/src/features/question-visibility-watchdog/watchdog.ts:85]: [sisyphus-bot] This schedules an unbounded 30-second timer for every question but the returned watchdog is never registered with `createPluginDispose`. The existing plugin lifecycle explicitly disposes managers/hooks; leaving this timer out means hot reload or server shutdown retains the closure and client reference, and can produce a toast from a dead plugin instance.

File: packages/omo-opencode/src/features/question-visibility-watchdog/watchdog.ts [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+  if (!isRecord(part) || part.type !== "tool") return false
+  const toolName = typeof part.tool === "string" ? part.tool : undefined
+  if (!toolName || !QUESTION_TOOL_NAMES.has(toolName.toLowerCase())) return false
+  if (callID === undefined) return true
+  return part.callID === callID
+}
+
+function partIsAwaitingUser(part: unknown): boolean {
+  if (!isRecord(part)) return false
+  if (!isRecord(part.state)) return true
```
Comment (code-yeongyu) [inline packages/omo-opencode/src/features/question-visibility-watchdog/watchdog.ts:141]: [sisyphus-bot] Missing `state` is classified as waiting. That is an unsafe default for a loosely typed transcript response: a partially hydrated/unknown tool part will produce a false reminder after 30 seconds. Require a recognized pending status (and explicitly handle the real OpenCode statuses) instead of turning malformed data into a user-visible alert.

File: packages/omo-opencode/src/features/question-visibility-watchdog/watchdog.ts [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+
+function messageParts(message: unknown): unknown[] {
+  if (!isRecord(message)) return []
+  const parts = message.parts
+  return Array.isArray(parts) ? parts : []
+}
+
+function partMatchesQuestionCall(part: unknown, callID: string | undefined): boolean {
+  if (!isRecord(part) || part.type !== "tool") return false
+  const toolName = typeof part.tool === "string" ? part.tool : undefined
```
Comment (code-yeongyu) [inline packages/omo-opencode/src/features/question-visibility-watchdog/watchdog.ts:133]: [sisyphus-bot] The probe only recognizes `part.tool` and `part.type === "tool"`, while the surrounding repository deliberately supports `tool_use`/`tool-invocation` and alternate `name`/`toolName` shapes for transcript compatibility. A server/client version that returns one of those shapes makes the watchdog report `resolved` and silently loses the very prompt this fix targets. Reuse the repository's normalization or add compatibility tests for the actual SDK response variants.

File: packages/omo-opencode/src/plugin-interface.ts [hunk 5]
```diff
@@ -33,6 +37,21 @@ export function createPluginInterface(args: {
   const { ctx, pluginConfig, firstMessageVariantGate, managers, hooks, tools } =
     args
 
+  const questionVisibilityWatchdog = createQuestionVisibilityWatchdog({
```
Comment (code-yeongyu) [inline packages/omo-opencode/src/plugin-interface.ts:40]: [sisyphus-bot] The watchdog is constructed here but never exposed to or called from the existing disposal path (`createPluginDispose`). This is not just an omitted unit test: production `dispose` will leave scheduled callbacks alive, and the callback closes over `ctx.client`/`ctx.directory`. Add the watchdog disposer to the lifecycle and test `pluginHooks.dispose()` cancels a pending check.

File: packages/omo-opencode/src/plugin/tool-execute-before.ts [hunk 6]
```diff
@@ -115,6 +117,9 @@ export function createToolExecuteBeforeHandler(args: {
       || normalizedToolName === "askuserquestion"
     ) {
       const sessionID = input.sessionID || getMainSessionID()
+      if (sessionID) {
+        questionVisibilityWatchdog?.onQuestionExecuted({ sessionID, callID: input.callID })
```
Comment (code-yeongyu) [inline packages/omo-opencode/src/plugin/tool-execute-before.ts:121]: [sisyphus-bot] This passes `input.callID` straight through, but the same repository's after-hook explicitly accepts `callID`, `callId`, and `call_id` because runtime integrations vary. If this seam receives a legacy casing or no call ID, the probe either misses the part or falls back to the first question in the session; concurrent/older questions can therefore get the wrong state and toast. Normalize the call ID at this boundary and test the fallback.

File: packages/omo-opencode/src/shared/markdown-link-audit.test.ts [hunk 7]
```diff
@@ -13,7 +13,13 @@ const MAINTAINER_LOCAL_PATH_RE = /file:\/\/\/(?:Users|home)\/|(?:^|[\s(`'"])(?:\
 function collectMarkdownFiles(): string[] {
   const output = Bun.spawnSync(["git", "ls-files", "*.md"], { cwd: WORKSPACE_ROOT, stdout: "pipe" })
   expect(output.exitCode).toBe(0)
-  return output.stdout.toString("utf-8").trim().split("\n").filter(Boolean).map((filePath) => resolve(WORKSPACE_ROOT, filePath))
+  return output.stdout
+    .toString("utf-8")
+    .trim()
+    .split("\n")
+    .filter(Boolean)
+    .filter((filePath) => !filePath.startsWith(".omo/") && !filePath.startsWith(".opencode/"))
```
Comment (code-yeongyu) [inline packages/omo-opencode/src/shared/markdown-link-audit.test.ts:21]: [sisyphus-bot] This excludes every tracked markdown file under `.omo/` and `.opencode/`, not merely the generated evidence paths that motivated the change. It turns off the audit for entire repository subtrees and allows broken links/local maintainer paths there to regress silently. Narrow the exclusion to the generated files (or filter only the known machine-generated link-bearing logs) and add a regression fixture proving a bad link in those directories is still caught.

File: packages/omo-opencode/src/features/question-visibility-watchdog/watchdog.test.ts [hunk 8]
```diff
@@ -0,0 +1,242 @@
...
+    expect(toasts).toHaveLength(1)
+    expect(toasts[0]).toContain("ses_parent")
+    watchdog.dispose()
+  })
+
+  test("#given question resolves within grace #when grace elapses #then no toast is shown", async () => {
+    const scheduler = createManualScheduler()
+    const toasts: string[] = []
+    const watchdog = createQuestionVisibilityWatchdog({
+      probeQuestionToolState: async () => resolvedProbe(),
```
Comment (code-yeongyu) [inline packages/omo-opencode/src/features/question-visibility-watchdog/watchdog.test.ts:70]: [sisyphus-bot] This test is mislabeled and tautological: the probe returns `resolved` only when the synthetic grace callback runs, so it does not establish that resolution happened before grace or that an early resolution cancels/suppresses the scheduled check. Add a controllable scheduler/settled signal that resolves before `runAll`, then assert no probe/toast occurs (or implement actual completion cancellation).

**Review by code-yeongyu — CHANGES_REQUESTED** [review]
[sisyphus-bot] REQUEST CHANGES. This patch adds a timer and a full session transcript fetch to a hot tool-execution seam without integrating cleanup or validating the production lifecycle. Several edge cases can produce false positives or stale toasts, while the user-facing recovery text does not match the actual question reply protocol. The added tests mostly validate isolated mocks and do not pin the real plugin wiring or lifecycle.

- The watchdog is never disposed by the plugin's existing dispose path, so unloaded/reloaded plugin instances retain timers and can call stale clients and show toasts after shutdown.
- Every question schedules a 30-second full `session.messages` fetch, including when no TUI exists; under many concurrent sessions this is avoidable network/load amplification, and there is no cancellation or timeout.
- The probe treats malformed/missing tool state as waiting and silently treats missing/mismatched transcript data as resolved; both choices can hide real state transitions or emit incorrect reminders.
- The toast instructs users to answer with a normal chat message even though the evidence and OpenCode protocol use the pending-question reply endpoint; this needs an end-to-end assertion that the suggested action actually unblocks the question.
- The markdown audit now excludes every `.omo/` and `.opencode/` markdown file rather than narrowly handling generated/local artifacts, weakening a repository-wide invariant.
- The "resolves within grace" test does not test resolution within the grace window: it only returns `resolved` when the synthetic scheduler fires, so it would pass with no early-resolution behavior at all.
